### PR TITLE
Add passphrase_file to mount options (refactored)

### PR DESCRIPTION
**Pull request by RlndVt**
Expansion of @donmor's work in #266

### CODE REVIEW

**Comment by donmor** [comment]
Reviewed. Now we may take a break and wait for any of our PRs to get merged :)

**Comment by JohnRTitor** [comment]
@RlndVt could you rebase this?

**Comment by RlndVt** [comment]
Thanks for the heads up, I've started looking into it.

File: src/commands/mount.rs [hunk 1]
```diff
@@ -373,6 +371,37 @@ fn cmd_mount_inner(opt: Cli) -> Result<()> {
...
+                error!(
+                    "Failed to read passphrase from file, falling back to prompt: {}",
+                    e
+                );
+                None
+            }
+        }
+    )
+    // Unlock by unlock policy
+    .unwrap_or_else(|| unlock_policy.apply(&first_sb)))
```
Comment (RlndVt) [inline src/commands/mount.rs:402]: Not completely happy with this flow yet.

The opening bracket on 389 matches the last closing bracket at the end of this line (402).  Preferably I would close it on line 400, but haven't figured out how to get the types matched up quite yet.

**Comment by RlndVt** [comment]
I'm not sure I understand the failing check; I believe "treefmt-check" is unhappy with the format of the code?

**Comment by RlndVt** [comment]
Sorry for the force-push spam. `treefmt-check` is a fickle beast.

**Review by JohnRTitor — COMMENTED** [review]
Nice, but @koverstreet should take a look.

**Comment by RlndVt** [comment]
In response to asking Kent to take a look:

On 19/05/2024 18:33, Kent Overstreet kent.overstreet@linux.dev wrote:
> I'm still catching up from conferences, but - it's not my preferred
> approach since it's easy to misuse, I know there's legitimate use cases
> but this makes it particularly easy for users to dumb things and leave
> their systems easy to compromise.
>
> Could you chat with Tony about other options? There's a couple (safer)
> things we still need, perhaps one of them will do what you need.

On Wednesday, June 5th, 2024 at 3:04 PM, Kent Overstreet <kent.overstreet@linux.dev> wrote:
> I'm not going to merge this; I've seen plenty of suggesteniosn for
> alternative workflows elsewhere.

I've asked (him and Tony) what the other options/alternative workflows are but no response yet.

Conclusion for this PR is still the same: won't be merged. Same (most likely) goes for #266.

**Comment by RlndVt** [comment]
A gist on how I mount my encrypted array at boot:

https://gist.github.com/RlndVt/7055be264c9492064d3523c8e74ea0a3

**Pull request closed without merging**